components/org.wso2.carbon.event.outbound.adapter.websubhub/src/main/java/org/wso2/carbon/event/outbound/adapter/websubhub/util/WebSubHubEventAdapterUtil.java

```diff
@@ -0,0 +1,303 @@
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonString;
+        try {
+            jsonString = mapper.writeValueAsString(securityEventTokenPayload);
+            request.setEntity(new StringEntity(jsonString));
+        } catch (IOException e) {
+            throw handleClientException(ERROR_PUBLISHING_EVENT_INVALID_PAYLOAD);
+        }
+
+        CloseableHttpAsyncClient client;
+        try {
+            client = WebSubHubEventAdapterDataHolder.getInstance().getClientManager()
+                    .getClient(tenantDomain);
+        } catch (IOException e) {
+            throw handleServerException(ERROR_GETTING_ASYNC_CLIENT, e, tenantDomain);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Publishing event data to WebSubHub. URL: " + url + " tenant domain: "
+                    + tenantDomain);
+        }
+
+        client.execute(request, new FutureCallback<HttpResponse>() {
+            @Override
+            public void completed(final HttpResponse response) {
+
+                int responseCode = response.getStatusLine().getStatusCode();
+                if (log.isDebugEnabled()) {
+                    log.debug("WebSubHub request completed. Response code: " + responseCode);
+                }
+                if (responseCode == 200 || responseCode == 201 || responseCode == 202 || responseCode == 204) {
+                    // Check for 200 success code range.
+                    if (log.isDebugEnabled()) {
+                        String jsonString;
+                        try {
+                            jsonString = EntityUtils.toString(response.getEntity());
+                            JSONParser parser = new JSONParser();
+                            JSONObject json = (JSONObject) parser.parse(jsonString);
+                            log.debug("Response data: " + json);
+                        } catch (IOException | ParseException e) {
+                            log.error("Error while reading WebSubHub event publisher response. " + e);
+                        }
+                    }
+                } else {
+                    log.error("WebHubSub event publisher received " + responseCode + " code.");
+                    String jsonString;
+                    try {
+                        jsonString = EntityUtils.toString(response.getEntity());
+                        JSONParser parser = new JSONParser();
+                        JSONObject json = (JSONObject) parser.parse(jsonString);
+                        log.error("Response data: " + json);
+                    } catch (IOException | ParseException e) {
+                        log.error("Error while reading WebSubHub event publisher response. " + e);
+                    }
+                }
+            }
+
+            @Override
+            public void failed(final Exception ex) {
+
+                log.error("Publishing event data to WebSubHub failed. " + ex);
```

Comment:
```suggestion
                log.error("Publishing event data to WebSubHub failed.", ex);
```

Reply:
resolve in https://github.com/wso2-extensions/identity-outbound-adapters/pull/1/commits/3c2955af8569cd206de3fe91be71d6362ed9160e